### PR TITLE
Account sign in

### LINKED ISSUE #6: Account sign-in

### PR DESCRIPTION
**Pull request by momoaolig**
## ✍️ Describe your changes

1. FInish account sign in ( the secret key used to generate jwt token is preset as 123456 ) .
2. Add the `AccountCreateResponse `class for signup function to return an insensitive account information.
3. **this branch is based on Branch Account-signup**

## 🎟 ️Issue ticket number and link
closes #6 

## ✅ Checklist before requesting a review
- [X] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: app/routers/account.py [hunk 1]
```diff
@@ -4,10 +4,17 @@
...
+secured_router = APIRouter(dependencies=[Depends(check_api_token)])
 
 
 @router.post("/")
 async def create_account(dto: AccountCreate, db: Session = Depends(get_db)):
     return await account_service.create_account(dto, db)
+
+
+@router.post("/signin")
+async def account_signin(dto: AccountCreate, db: Session = Depends(get_db)):
```
Comment (yhware) [inline app/routers/account.py:19]: dto should be different. Create a dedicated dto for signin. It should have email and password.

File: app/services/account.py [hunk 2]
```diff
@@ -19,7 +21,43 @@ async def create_account(dto: AccountCreate, db: Session):
...
+    hashed_pw = Hasher.get_password_hash(dto.password)
+    account = (
+        db.query(Account)
+        .filter(
+            Account.email == dto.email, Hasher.verify_password(dto.password, hashed_pw)
+        )
+        .first()
+    )
+    if account is None:
+        # raise HTTPException(status_code=401, detail="Invalid credentials")
```
Comment (yhware) [inline app/services/account.py:49]: remove unnecessary comment

File: app/services/account.py [hunk 3]
```diff
@@ -19,7 +21,43 @@ async def create_account(dto: AccountCreate, db: Session):
...
+        return {"error": "wrong password"}
+
+    secret_key = "123456"
```
Comment (yhware) [inline app/services/account.py:52]: get the secret key through an environment var

File: app/services/account.py [hunk 4]
```diff
@@ -19,7 +21,43 @@ async def create_account(dto: AccountCreate, db: Session):
...
+    algo = "HS256"
+    payload = {
+        "sub": account.email,
+        "exp": time.time() + 7200,
```
Comment (yhware) [inline app/services/account.py:56]: maybe change the valid time to sth longer like 8 hours? 2 hours seem a little too short.

**Review by yhware — CHANGES_REQUESTED** [review]

File: app/services/account.py (hunk 2 above)
Comment (momoaolig) [reply app/services/account.py:49]: want to make sure is that ok now to return error directly, or to use this comment line to raise a http exception

File: app/routers/account.py (hunk 1 above)
Comment (yhware) [reply app/routers/account.py]: This is still an issue

File: app/services/account.py [hunk 5]
```diff
@@ -19,7 +22,42 @@ async def create_account(dto: AccountCreate, db: Session):
...
-    return new_account
+
+    return new_account_response
+
+
+async def account_signin(dto: AccountCreate, db: Session):
+    if not get_by_email(db, dto.email):
+        return {"error": "the account does not exist"}
+
+    hashed_pw = Hasher.get_password_hash(dto.password)
```
Comment (yhware) [inline app/services/account.py:41]: Verifying password logic is wrong.
You should first retrieve the account from the database and use verify_password function to check if the plaintext password given by the user is the same as the hashed password in the database.

In the current logic, hashed_pw will always be a different value due to salting.

**Review by yhware — CHANGES_REQUESTED** [review]
Except for the comment about pw checking, it all looks good.

**Pull request merged**